mountpoint-s3-client/src/endpoint_config.rs

```diff
@@ -148,10 +152,57 @@ impl EndpointConfig {
         let resolved_endpoint = endpoint_rule_engine
             .resolve(endpoint_request_context)
             .map_err(EndpointError::UnresolvedEndpoint)?;
-        let endpoint_uri = resolved_endpoint.get_url();
-        Uri::new_from_str(&allocator, endpoint_uri)
+
+        Ok(ResolvedEndpointInfo(resolved_endpoint))
+    }
+}
+
+#[derive(Debug, Clone)]
+pub struct ResolvedEndpointInfo(ResolvedEndpoint);
+
+impl ResolvedEndpointInfo {
+    pub fn get_uri_from_endpoint(&self) -> Result<Uri, EndpointError> {
+        let allocator = Allocator::default();
+        let endpoint_url = self.0.get_url();
+        Uri::new_from_str(&allocator, endpoint_url)
             .map_err(|e| EndpointError::InvalidUri(InvalidUriError::CouldNotParse(e)))
     }
+
+    pub fn get_auth_scheme_from_endpoint(&self) -> Result<AuthScheme, EndpointError> {
+        let endpoint_properties = self.0.get_properties();
+        let auth_scheme_data: serde_json::Value = serde_json::from_slice(endpoint_properties.as_bytes())?;
+        let auth_scheme_value = auth_scheme_data
+            .get("authSchemes")
+            .ok_or_else(|| EndpointError::InvalidAuthScheme("AuthScheme is missing from properties".to_owned()))?;
+        let disable_double_encoding = auth_scheme_value
+            .get("disableDoubleEncoding")
+            .and_then(|t| t.as_bool())
+            .ok_or_else(|| {
+                EndpointError::InvalidAuthScheme("disableDoubleEncoding is missing from authScheme json.".to_owned())
+            })?;
+        let scheme_name = auth_scheme_value
+            .get("name")
+            .and_then(|t| t.as_str())
+            .ok_or_else(|| EndpointError::InvalidAuthScheme("name is missing from authScheme json.".to_owned()))?;
+        let signing_name = auth_scheme_value
+            .get("signingName")
+            .and_then(|t| t.as_str())
+            .ok_or_else(|| {
+                EndpointError::InvalidAuthScheme("signingName is missing from authScheme json.".to_owned())
+            })?;
+        let signing_region = auth_scheme_value
+            .get("signingRegionSet")
+            .and_then(|t| t.as_str())
+            .ok_or_else(|| {
+                EndpointError::InvalidAuthScheme("signingRegionSet is missing from authScheme json.".to_owned())
+            })?;
```

Comment:
Are all five of these fields always present -- is failing the right thing to do if any of them are missing?

I think you were saying earlier that not every resolution result includes this structure at all, so maybe this should return `Result<Option<AuthScheme>, ...>`?

Reply:
Like the region is set as '*' when it is not part of ARN. Not found any case where they are missing in testcases. Need to test normal bucket and aliases.